_posts/2021-04-21-3d-bonet.md

```diff
@@ -0,0 +1,218 @@
+</div>
+
+The architecture is very similar to a traditional architecture used on images: backbone (feature extractor) and different heads for doing multi-taks learning. Each head has a specific goal.
+
+Given an input point cloud $$\textbf{P}$$ with $$N$$ points in total $$\textbf{P} \in \mathbb{R}^{N \times k_0} $$ where $$ k_0 $$ is the number of channels such as the location $$\{x,y,z\}$$ and color $$\{r,g,b\}$$ of each point, the backbone network extract **point local features** $$\textbf{F}_l \in \mathbb{R}^{N \times k}$$, and aggregates a **global point cloud feature vector** $$\textbf{F}_g \in \mathbb{R}^{1 \times k}$$, where $$k$$ is the lenght of feature vectors.
+
+The bounding box prediction branch takes $$\textbf{F}_g$$ as input and regress a **predefined and fixed set of bounding boxes** $$B$$ and the corresponding box scores $$B_s$$.
+
+During the training this branch is supervised with ground truth bounding bboxes and fed into a **box association layer** that automatically associates a unique and most similar predicted bounding box to each ground truth box.
+
+The output of the association layer is a list of association index $$\textbf{A}$$.
+
+Using the indices, the predicted bboxes are reorganized such that each gt box is paired with only one predicted box for the loss calculation. The predicted boxes are also reordered accordingly (the score) before the loss calculation.
+
+**note**: the bounding box association layer and multi-criteria loss function are only designed for network training - discarded, thus, during testing.
+
+Every predicted box + the local and global features ($$\textbf{F}_l$$ and $$\textbf{F}_g$$) are fed into the **point mask prediction branch** -> this branch is shared by all instances of difference categories, and therefore extremely light and compact. This is a **class agnostic** approach that allows general segmentation across **unseen categories**.
+
+
+### Bounding Box Prediction
+
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="bbox prediction" src="/images/3dbonet/3d-bo-net-bbox.png">
+    <figcaption> The Bounding box predictiond module </figcaption>
+    </figure>
+</div>
+
+Simple parametrization of bbox: $$ \{ [x_\text{min}, y_\text{min}, z_\text{min}], [x_\text{max}, y_\text{max}, z_\text{max}]\} $$
+
+#### Neural architecture
+
+The global feature vector $$\textbf{F}_g$$ is fed through 2 fully connected layers with Leaky ReLU, followed by 2 parallel FC layers.
+
+- 1 layer outputs a 6H dimensional vector, reshaped into a $$ H \times 2 \times 3$$ tensor, where H is the **predefined and fixes number of bounding boxes**
+- 1 layer outputs a H dimensional vector followed by a **sigmoid** to represent the bounding box scores.
+
+
+**Bounding Box Association Layer**
+
+Given the H predicted bboxes $$ \textbf{B} \in \mathbb{R}^{H \times 2 \times 3} $$ how to use the gt bboxes $\textbf{\bar{B}} \in \mathbb{R}^{T \times 2 \times 3}$$ to supervise the network?
```

Comment:
At the beginning of the second formula, a `$` is missing.
Moreover, I don't know if I have some issues with my configuration, but the second formula doesn't display correctly, probably the `\bar` goes before the `\textbf` :
` $$ \bar{\textbf{B}} \in \mathbb{R}^{T \times 2 \times 3}  $$ `